extras/asm-multicluster/cluster-setup.sh

```diff
@@ -0,0 +1,149 @@
+    exit 1
+fi
+
+export CLUSTER_1_NAME="cluster-1"
+export CLUSTER_1_ZONE="us-central1-a"
+
+export CLUSTER_2_NAME="cluster-2"
+export CLUSTER_2_ZONE="us-central1-b"
+############################################################
+
+
+# Note - workload identity is required for ASM 
+echo "☸️ Creating clusters..."
+
+gcloud config set project ${PROJECT_ID}
+gcloud services enable container.googleapis.com
+
+gcloud beta container clusters create ${CLUSTER_1_NAME} \
+--project=${PROJECT_ID} --zone=${CLUSTER_1_ZONE} \
+--release-channel=regular  --image-type cos_containerd \
+--machine-type=e2-standard-4 --num-nodes=4 \
+--workload-pool=${PROJECT_ID}.svc.id.goog --async 
+
+gcloud beta container clusters create ${CLUSTER_2_NAME} \
+--project=${PROJECT_ID} --zone=${CLUSTER_2_ZONE} \
+--release-channel=regular  --image-type cos_containerd \
+--machine-type=e2-standard-4 --num-nodes=4 \
+--workload-pool=${PROJECT_ID}.svc.id.goog 
+
+
+echo "💻 Setting local kubectx..."
+gcloud container clusters get-credentials ${CLUSTER_1_NAME} --zone ${CLUSTER_1_ZONE} 
+kubectx cluster-1=. 
+
+gcloud container clusters get-credentials ${CLUSTER_2_NAME} --zone ${CLUSTER_2_ZONE} 
+kubectx cluster-2=. 
+
+
+echo "⬇️ Installing required tools for ASM..."
+# https://cloud.google.com/service-mesh/docs/scripted-install/asm-onboarding#installing_required_tools 
+curl https://storage.googleapis.com/csm-artifacts/asm/install_asm_1.9 > install_asm
+curl https://storage.googleapis.com/csm-artifacts/asm/install_asm_1.9.sha256 > install_asm.sha256
+sha256sum -c --ignore-missing install_asm.sha256 
+chmod +x install_asm
+
+echo "🕸 Installing ASM on cluster-1..."
+# https://cloud.google.com/service-mesh/docs/managed-control-plane 
+kubectx cluster-1 
+
+./install_asm --mode install --managed -p ${PROJECT_ID} \
+    -l ${CLUSTER_1_ZONE} -n ${CLUSTER_1_NAME} -v \
+    --output_dir asm-${CLUSTER_1_NAME} --enable-all 
+
+# get "revision" from directory created by install_asm script 
+for dir in asm-${CLUSTER_1_NAME}/istio-*/ ; do
+    export REVISION=`basename $dir`
+    echo "Revision is: $REVISION" 
+done
+
+echo "⛵️ Installing the Istio IngressGateway on cluster-1..."
+./asm-${CLUSTER_1_NAME}/$REVISION/bin/istioctl install -f ./asm-${CLUSTER_1_NAME}/managed_control_plane_gateway.yaml --set revision=asm-managed -y
+
+echo "🕸 Installing ASM on cluster-2..."
+kubectx cluster-2 
+
+./install_asm --mode install --managed -p ${PROJECT_ID} \
+    -l ${CLUSTER_2_ZONE} -n ${CLUSTER_2_NAME} -v \
+    --output_dir asm-${CLUSTER_2_NAME} --enable-all
+
+
+echo "🌏 Setting up Endpoint Discovery between clusters..."
+# https://cloud.google.com/service-mesh/docs/managed-control-plane#configure_endpoint_discovery_only_for_multi-cluster_installations 
+
+export CTX_1="cluster-1"
+export CTX_2="cluster-2"
+
+echo "Letting cluster 1 know about cluster 2..."  
+./asm-${CLUSTER_1_NAME}/$REVISION/bin/istioctl x create-remote-secret --context=${CTX_2} --name=cluster2 | kubectl apply -f - --context=${CTX_1}
```

Comment:
I think the names for the secrets here must match the actual cluster names as per the notes [here](https://istio.io/latest/docs/ops/diagnostic-tools/multicluster/#step-by-step-diagnosis)
```sh
--name=${CLUSTER_2_NAME} # for the secret in the first cluster
--name=${CLUSTER_1_NAME} # for the secret in the second cluster
```

Reply:
fixed!